### ISSUE TITLE
wandb 0.8.0 stops reporting by first eval

### ISSUE THREAD

**Issue by mloenow** [wandb/client#346]
* Weights and Biases version: 0.8.0
* Python version: 3.6.7
* Operating System: Ubuntu 18.04 (Google Colab, Python 3 GPU runtime)

### Description

I'm fine-tuning a model with a custom dataset using the TensorFlow Object Detection API. All data and metadata is stored and saved on Google Cloud Storage (GCS), incl. the TensorBoard metadata/logs. Since version 0.8.0, reporting stops after a little while - seemingly when the model is being evaluated.

### What I Did

`!pip install wandb`
`!wandb login KEY`

A modified `model_main.py` is then executed:

```
!python object_detection/model_main_wandb.py \
  --pipeline_config_path="${CONFIGS_DIR}/${PIPELINE_FILE}" \
  --model_dir="${TRAIN_DIR}"
```

Which runs this method before training starts:

```
def wandb_init():
  # Get pipeline config in a W&B friendly format
  pipeline_config = pipeline_pb2.TrainEvalPipelineConfig()
  with tf.gfile.GFile(FLAGS.pipeline_config_path, "r") as f:
    proto_str = f.read()
    text_format.Merge(proto_str, pipeline_config)

  wandb_config = json_format.MessageToDict(message=pipeline_config)
  wandb.init(config=wandb_config, tensorboard=True)
```

It does report files that it's unable to access on GCS, but that also happened before and it still worked: `wandb: gs://BUCKET/v1/runs/RUN_ID/eval_0/events.out.tfevents.1558516485.622e141833f4 is a cloud storage url, can't save file to wandb.` (for 0.7.3 - let me know if I should reproduce for 0.8.0, but they are similar)

This works without issues on 0.7.3 (I've now pinned this version to confirm), even with the error/warning above, except for some files being unavailable on W&B, e.g. the checkpoints, but I was fine with just the metrics. I can see a "TensorBoard Watcher" was introduced in 0.8.0 - perhaps it's related to this?

Let me know if you need some more details from me.

**Comment by vanpelt** [comment]
Hey @mloenow, what version of tensorflow are you using?  Can you link to a partially finished run?  If you can access the wandb/debug.log that would help as well.

**Comment by mloenow** [comment]
Hey @vanpelt, I'm using TensorFlow version 1.13.1. I'll send the link on Intercom now, although it has since been resumed with wandb version 0.7.3. The runtime was reset before it was resumed, so wandb/debug.log won't have anything with version 0.8.0 in it. Let me reproduce it with a new run and I'll send you a link to and the log file after it stops reporting properly for a while.

**Comment by mloenow** [comment]
@vanpelt - I've sent you a link to a run with the issue, as well as the debug.log file. Thanks!

**Comment by vanpelt** [comment]
Hey @mloenow thanks for the info.  Unfortunately the logs didn't provide me much about the root cause.  Is there any chance you could share the code with me so I can try to reproduce locally?  I did add some fixes for the error messages you were getting about writing the images.  You can try it by running:  `pip install --upgrade git+git://github.com/wandb/client.git@master#egg=wandb`.  I don't think it solved the root cause but it may have.

**Comment by mloenow** [comment]
@vanpelt - `master` actually works! I've sent you a link to the new run. Any idea what the cause was? One thing I noticed right away was the `eval_0` namespace being correctly prepended instead of appended (without slash) before (`eval_0/DetectionBoxes_Precision/mAP` vs. `DetectionBoxes_Precision/mAPeval_0`).

If you still want to reproduce it locally, I was running this beast: https://github.com/tensorflow/models/tree/master/research/object_detection. I modified `model_main.py` to initialize `wandb`, as initially mentioned. The pipeline configuration file was updated to use GCS endpoints (`train_config.fine_tune_checkpoint`, `train_input_reader.label_map_path`, `train_input_reader.tf_record_input_reader.input_path`, `eval_input_reader.label_map_path`, `eval_input_reader.tf_record_input_reader.input_path`). I can send you the relevant cell of my notebook to help if needed.

Thanks for your assistance!

BTW - will GCS be supported at some point? In my case I'm authenticating with `from google.colab import auth; auth.authenticate_user()`, but could be a credentials file instead - with that, I believe anything executing on the runtime would be able to pull data from the GCS bucket, so perhaps the client could retrieve data there using `gsutil` or similar.
